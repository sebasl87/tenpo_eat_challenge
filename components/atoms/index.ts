export { default as RestaurantsLayout } from './RestaurantsLayout';
