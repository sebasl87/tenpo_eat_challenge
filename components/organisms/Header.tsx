import React from 'react'
import { Image, Text, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.header}>
                <Image
                    source={require("../../assets/imgs/user.png")}
                    style={styles.userImg}
                />
                <Image
                    source={require("../../assets/icons/search.png")}
                    style={styles.searchImg}
                />
            </View>
            <View style={styles.header}>
                <View style={{ flexDirection: "column" }}>
                    <Image
                        source={require("../../assets/imgs/logo.png")}
                        style={styles.logoImg}
                    />
                    <Text>DELIVER APP</Text>
                </View>
                <View style={{ flexDirection: "row", marginBottom: -16 }}>
                    <Image
                        source={require("../../assets/imgs/hand.png")}
                        style={styles.handImg}
                    />
                    <View style={{ top: "100%", height: 6 }}>

                        <Image
                            source={require("../../assets/icons/ellipse.png")}
                            style={styles.ellipseImg}
                        />
                    </View>
                </View>
            </View>
        </>
    );
}

export default Header