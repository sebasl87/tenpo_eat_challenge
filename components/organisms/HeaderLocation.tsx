import React from 'react'
import { Image, Text, TextInput, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.headerLocation}>
                <View style={{ marginBottom: 32, display: "flex", flexDirection: "row" }}>
                    <Image
                        source={require("../../assets/icons/location.png")}
                        style={{ height: 24, width: 24, marginRight: 6, paddingTop: 30 }}
                    />
                    <Text style={styles.fontTitle}>Agregar dirección de la entrega</Text>
                </View>
            </View>
            <TextInput placeholder='Escribe tu dirección' style={styles.inputLocation} />
        </>
    );
}

export default Header