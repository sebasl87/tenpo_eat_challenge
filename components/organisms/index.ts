export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
