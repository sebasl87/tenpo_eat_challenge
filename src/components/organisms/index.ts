export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
