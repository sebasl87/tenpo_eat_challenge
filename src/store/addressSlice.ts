import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import {  Location } from './googleSlice';

export interface IAddress {
  formated_address: string | undefined;
  extra_data_address?: string | undefined;
  geolocation?: Location | undefined;
}

const initialAddressState = {
  address: {} as IAddress,
};

export const addressSlice = createSlice({
  name: 'address',
  initialState: initialAddressState,
  reducers: {
    setAddress(state, action: PayloadAction<IAddress>) {
      state.address = action.payload;
    },
  },
});

export const { setAddress } = addressSlice.actions;
