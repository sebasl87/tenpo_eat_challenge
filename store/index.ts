import { configureStore } from '@reduxjs/toolkit';

import { googleSlice } from './googleSlice';

export const store = configureStore({
  reducer: {
    google: googleSlice.reducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
