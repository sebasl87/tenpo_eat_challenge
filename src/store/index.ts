import { configureStore } from '@reduxjs/toolkit';

import { googleSlice } from './googleSlice';
import { addressSlice } from './addressSlice';
import { restaurantsSlice } from './restaurantsSlice';

export const store = configureStore({
  reducer: {
    google: googleSlice.reducer,
    address: addressSlice.reducer,
    restaurants: restaurantsSlice.reducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
