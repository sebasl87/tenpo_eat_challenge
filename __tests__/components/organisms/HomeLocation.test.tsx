import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { HomeLocation } from "../../../src/components/organisms";
import { IAddress } from "../../../src/store/addressSlice";

export const mock_address: IAddress = {
    formated_address: "Some mock street, Santiago, Chile",
    extra_data_address: "Mocked extra data",
    geolocation: { lat: -70, lng: -34 }
}

describe("HomeLocation component", () => {
    it("should render successfully with HomeLocation_mock", () => {
        const homeLocationSection = renderer
            .create(<HomeLocation address={mock_address} />)
            .toJSON();

        expect(homeLocationSection).toMatchSnapshot();
    });
});
