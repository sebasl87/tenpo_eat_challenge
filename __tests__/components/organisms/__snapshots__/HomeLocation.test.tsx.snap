// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeLocation component should render successfully with HomeLocation_mock 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#D4F9F5",
        "borderTopLeftRadius": 28,
        "borderTopRightRadius": 28,
        "paddingBottom": 18,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 20,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/location.png",
          }
        }
        style={
          {
            "height": 18,
            "marginRight": 6,
            "paddingTop": 30,
            "width": 24,
          }
        }
      />
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#008f7e",
              "fontSize": 16,
              "marginLeft": 8,
              "maxWidth": 300,
            },
            {
              "fontFamily": "Gotham-Light",
            },
          ]
        }
      >
        Some mock street, Santiago, Chile
      </Text>
    </View>
  </View>
</View>
`;
